api/libretime_api/storage/views/file.py

```diff
@@ -29,3 +29,19 @@ def download(self, request, pk=None):  # pylint: disable=invalid-name
         redirect_uri = filepath_to_uri(os.path.join("/api/_media", file.filepath))
         response["X-Accel-Redirect"] = redirect_uri
         return response
+
+    @action(detail=True, methods=["DELETE"])
+    def delete_file(self, request, pk=None):  # pylint: disable=invalid-name
+        pk = IntegerField().to_internal_value(data=pk)
+
+        file = get_object_or_404(File, pk=pk)
+        path = file.filepath
+
+        try:
+            if os.path.isfile(path):
+                os.remove(path)
+        except OSError as exception:
+            logger.error(f"Could not delete file from storage: {exception}")
+            return HttpResponse(status=404)
```

Comment:
I think this should return a 500.

Reply:
I was debating that. 4040 wuld mean "i didn't find the file i was supposed to delete" while 500 means "the database entry is there but the file isn't, there's something fishy here". Fine with 500 as well :-)

Reply:
I think the behavior is a slightly different: we do check if the files exist first. So if the file is missing, we continue by deleting the database entry.

But if deleting the storage file fails because of e.g. permission denied error, then we have another problem.

So yes, we agree on the 500.